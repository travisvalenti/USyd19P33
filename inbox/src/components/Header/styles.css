.Header {
  background-color: dodgerblue;
  width: 100%;
  height: 60px;
  line-height: 60px;
}

.Header ul {
  list-style: none;
  margin: 0
}

.Header li {
  list-style: none;
  padding: 0 1em;
  float: left;
  box-sizing: border-box;
  border-radius: 0 0 3px 3px;
  margin-left: 10px;
}

.Header li:hover, .Header li.selected {
  background: #1a59d6;
  height: 63px;
}


.Header a {
  color: #ffffff;
  font-family: Arial;
  font-size: 150%;
  text-decoration: none;
  outline: 0;
}
