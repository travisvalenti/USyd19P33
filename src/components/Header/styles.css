.Header {
  background-color: dodgerblue;
  width: 100%;
  height: 60px;
  line-height: 60px;
}

.Header ul {
  list-style: none;
  margin: 0
}

.Header li {
  list-style: none;
  padding: 0 1em;
  float: left;
  box-sizing: border-box;
  border-radius: 0 0 3px 3px;
  margin-left: 1em;
  color: white;
}

.Header li:hover {
  color: #ddd;
}

.Header li.selected{
  background: #1a59d6;
  height: 63px;
  color: white;
}

.Header a {
  font-family: Arial;
  font-size: 150%;
  text-decoration: none;
  outline: 0;
}

.material-icons {
  position: relative;
  top: 3px;
}