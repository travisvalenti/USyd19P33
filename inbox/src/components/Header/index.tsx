import React from 'react'
import { Link, RouteComponentProps } from 'react-router-dom'

import './styles.css'

const Header = ({ location }: RouteComponentProps) => {
  console.log(location)
  return (
    <div className="Header">
      <ul>
        <Link to="/">
          <li className={location.pathname === '/' ? 'selected' : undefined}>
            Dashboard
          </li>
        </Link>
        <Link to="/mail">
          <li className={location.pathname === '/mail' ? 'selected' : undefined}>
            Mail
          </li>
        </Link>
      </ul>
    </div>
  )
}

export default Header