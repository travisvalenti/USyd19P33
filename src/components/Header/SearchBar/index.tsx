/** SearchBar
 * This component will allow the user to input a text string and it will
 * redirect to the specified search page on Docs, Sheets, etc.
 * It is designed to be placed within the Header.
 */
import React from 'react'
import { RouteComponentProps } from 'react-router-dom'

import './styles.css'

enum SearchType {
  Mail = "Search Mail",
  Google = "Search Google",
  Drive = "Search Drive",
  Gmail = "Search Gmail",
  Docs = "Search Docs",
  Sheets = "Search Sheets",
  Slides = "Search Slides"
}

type Props = {
  queryString : string
  onQueryChange : any
}

type State = {
  value : string
  searchType : SearchType
  displayStyle : any
}

class SearchBar extends React.Component<Props & RouteComponentProps, State> {


  constructor(props : Props & RouteComponentProps) {
    super(props)

    this.state = {
      value : "",
      searchType : SearchType.Mail,
      displayStyle : { display: "none" }
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleDropdown = this.handleDropdown.bind(this);
  }

  handleDropdown(event : any) {
    if(this.state.displayStyle.display === "none") {
      this.setState({ displayStyle : {display: "block"} });
    }
    else {
      this.setState({ displayStyle : {display: "none"} });
    }
  }

  handleChange(event : any) {
    this.setState({value: event.target.value});
  }

  handleSubmit(event : any) {
    event.preventDefault();
    if(this.state.searchType === SearchType.Mail) {
      this.props.onQueryChange(this.state.value)
      this.props.history.push('/mail/read')
    }
    if(this.state.searchType === SearchType.Google) {
      window.location.assign("https://www.google.com/search?q=" + this.state.value);
    }
    if(this.state.searchType === SearchType.Drive) {
      window.location.assign("https://drive.google.com/drive/search?q=" + this.state.value);
    }
    if(this.state.searchType === SearchType.Gmail) {
      window.location.assign("https://mail.google.com/mail/u/0/#search/" + this.state.value);
    }
    if(this.state.searchType === SearchType.Docs) {
      window.location.assign("https://docs.google.com/document/u/0/?q=" + this.state.value);
    }
    if(this.state.searchType === SearchType.Sheets) {
      window.location.assign("https://docs.google.com/spreadsheets/u/0/?q=" + this.state.value);
    }
    if(this.state.searchType === SearchType.Slides) {
      window.location.assign("https://docs.google.com/presentation/u/0/?q=" + this.state.value);
    }
  }

  handleTypeChange(type : SearchType, event : any) {
    this.setState({searchType : type});
    if(this.state.displayStyle.display === "none") {
      this.setState({ displayStyle : {display: "block"} });
    }
    else {
      this.setState({ displayStyle : {display: "none"} });
    }
  }

  render() {
    return(
      <div className="SearchBar">
        <div className="dropdownArrow">
          <button onClick={this.handleDropdown}>
            <svg viewBox="0 0 24 24"><path d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>
          </button>
        </div>
        <div className="searchBarInput">
          <form onSubmit={this.handleSubmit}>
            <input type="text" placeholder={this.state.searchType} value={this.state.value} onChange={this.handleChange}/>
            <button type="submit"><svg><path d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"/><path d="M0 0h24v24H0z" fill="none"/></svg></button>
          </form>
        </div>
        <div className="dropdownMenu" style={this.state.displayStyle}>
          <p onClick={(e : any) => this.handleTypeChange(SearchType.Mail, e)}>Search Mail</p>
          <p onClick={(e : any) => this.handleTypeChange(SearchType.Google, e)}>Search Google</p>
          <p onClick={(e : any) => this.handleTypeChange(SearchType.Drive, e)}>Search Drive</p>
          <p onClick={(e : any) => this.handleTypeChange(SearchType.Gmail, e)}>Search Gmail</p>
          <p onClick={(e : any) => this.handleTypeChange(SearchType.Docs, e)}>Search Docs</p>
          <p onClick={(e : any) => this.handleTypeChange(SearchType.Sheets, e)}>Search Sheets</p>
          <p onClick={(e : any) => this.handleTypeChange(SearchType.Slides, e)}>Search Slides</p>
        </div>
      </div>
    )
  }
}

export default SearchBar