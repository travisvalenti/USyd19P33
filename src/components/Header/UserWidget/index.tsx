import React from 'react'

import './styles.css'

import Button from '../../ui/Button'

type Props = {
    isSignedIn : boolean
}

type State = {

}

class UserWidget extends React.Component<Props, State> {

    render() {
        let userProfile : any;
        if(this.props.isSignedIn) {
            userProfile = gapi.auth2.getAuthInstance().currentUser.get().getBasicProfile();
        }
        return (
            <div className="UserWidget">
                {this.props.isSignedIn
                    ? <div className="userButton"><Button className="buttonHeader" onClick={() => gapi.auth2.getAuthInstance().signOut()}>Sign Out</Button></div>
                    : <div className="userButton"><Button className="buttonHeader" onClick={() => gapi.auth2.getAuthInstance().signIn()}>Sign In</Button></div>
                }
                {this.props.isSignedIn &&
                    <div className="userProfile">
                        <div className="userName"><p>{userProfile.ig}</p></div>
                        <div className="userPhoto"><img src={userProfile.Paa} alt="User"/></div>
                    </div>
                }
            </div>
        )
    }
}

export default UserWidget