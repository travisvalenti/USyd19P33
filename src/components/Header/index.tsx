import React from 'react'
import { Link, RouteComponentProps } from 'react-router-dom'

import './styles.css'

import SearchBar from './SearchBar'
import UserWidget from './UserWidget'

type Props = {
  isSignedIn : boolean
  queryString : string
  onQueryChange : any
}

const Header = (props : Props & RouteComponentProps) => {

  return (
    <div className="Header">
      <div className="navigateItems">
        <ul>
          <Link to="/">
            <li className={props.location.pathname === '/' || props.location.pathname === '/search' ? 'selected' : undefined}>
              Dashboard
            </li>
          </Link>
          <Link to="/mail/write">
            <li className={props.location.pathname === '/mail/write' ? 'selected' : undefined}>
              <i className="material-icons">edit</i> Write
            </li>
          </Link>
          <Link to="/mail/read">
            <li className={props.location.pathname === '/mail/read' ? 'selected' : undefined}>
              <i className="material-icons">email</i> Read
            </li>
          </Link>
        </ul>
      </div>
      <SearchBar queryString={props.queryString} onQueryChange={props.onQueryChange}/>
      <UserWidget isSignedIn={props.isSignedIn} />
    </div>
  )
}

export default Header