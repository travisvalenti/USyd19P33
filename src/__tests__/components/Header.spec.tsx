import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { createMemoryHistory } from 'history'

import { Router, Route } from 'react-router-dom'

import Header from '../../components/Header'

describe('Header component', () => {
  afterEach(cleanup)

  // Note that this snapshot must be updated when the user-widget (#12) is implemented
  test('Matches the snapshot', () => {
    const component = render(
      <Router history={createMemoryHistory()}>
        <Route path="/" component={Header} />
      </Router>
    )
    expect(component.container).toMatchSnapshot()
  })

  test('Navigates to Write', () => {
    const history = createMemoryHistory()
    const component = render(
      <Router history={history}>
        <Route path="/" component={Header} />
      </Router>
    )
    const navButton = component.getByText('Write')
    fireEvent.click(navButton)
    expect(history.location.pathname).toBe('/mail/write')
  })

  test('Navigates to Read', () => {
    const history = createMemoryHistory()
    const component = render(
      <Router history={history}>
        <Route path="/" component={Header} />
      </Router>
    )
    const navButton = component.getByText('Read')
    fireEvent.click(navButton)
    expect(history.location.pathname).toBe('/mail/read')
  })

  // TODO: issue #12
  test.skip('Contains a user information section', () => {
    const history = createMemoryHistory()
    const component = render(
      <Router history={history}>
        <Route path="/" component={Header} />
      </Router>
    )
    expect(component.getByTestId('user-widget')).toBeTruthy()
  })
})
