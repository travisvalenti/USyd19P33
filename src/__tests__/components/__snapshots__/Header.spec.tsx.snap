// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches the snapshot 1`] = `
<div>
  <div
    class="Header"
  >
    <ul>
      <a
        href="/"
      >
        <li
          class="selected"
        >
          Dashboard
        </li>
      </a>
      <a
        href="/mail/write"
      >
        <li>
          <i
            class="material-icons"
          >
            edit
          </i>
           Write
        </li>
      </a>
      <a
        href="/mail/read"
      >
        <li>
          <i
            class="material-icons"
          >
            email
          </i>
           Read
        </li>
      </a>
    </ul>
  </div>
</div>
`;
